### ISSUE TITLE
Questions about `update_mask_approx2`

### ISSUE THREAD

**Issue by Nusselder9** [papers-submission/structured_transposable_masks#1]
Hi! I really appreciate the authors for doing such an inspiring job and sharing the code. But it seems that the results of 2- approximation algorithm may have some bugs.

I found that your implementation of 2-approximation algorithm at [here](https://github.com/papers-submission/structured_transposable_masks/blob/main/dynamic_TNM/src/sparse_ops.py#L22). Taking 2:4 Transposable Mask as an example, if the input is `[[1 2 3 4],[5 6 7 8],[9 10 11 12],[13 14 15 16]]`, the output of the above implementation will be `[[1 0 1 0],[0 1 1 0],[0 0 1 0],[1 1 1 1]]`, which does not satisfy 2:4 Transposable Sparsity.

Where did I go wrong?  Thanks!

**Comment by papers-submission** [comment]
You are right :) line 28 should be change in the case of 2:4 to k<2 and generally to k<topk. We experimented only with 4:8 as our equations (in the mask diversity section) suggest that 2:4-transposable would suffer from accuracy degradation.  I will push a fix to make this function more general.

**Comment by Nusselder9** [comment]
Thanks for you kind reply!
It seems that even if I test with 8*8 input, the bug still exists.

The test code is as below,
```
data=torch.arange(1,65).view(1,64)
mask=torch.zeros_like(data)
update_mask_approx2(data,mask, BS=8).view(8,8)
```
Then I get the result as  below,
```
tensor([[1, 0, 1, 1, 1, 0, 0, 0],
        [0, 1, 1, 1, 1, 0, 0, 0],
        [0, 0, 1, 1, 1, 0, 0, 0],
        [0, 0, 0, 1, 1, 0, 0, 0],
        [0, 0, 0, 0, 1, 0, 0, 0],
        [1, 1, 1, 1, 1, 1, 0, 0],
        [1, 1, 1, 1, 1, 1, 1, 0],
        [1, 1, 1, 1, 1, 1, 1, 1]])
```
It does not satisfy 4:8 Transposable Sparsity.

**Comment by papers-submission** [comment]
Fixed

**Comment by Nusselder9** [comment]
Thanks!
